cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -621,6 +622,12 @@ struct GroupedValueTraits {
 
   static CType Get(const CType* values, uint32_t g) { return values[g]; }
   static void Set(CType* values, uint32_t g, CType v) { values[g] = v; }
+  static Status AppendBuffers(TypedBufferBuilder<CType>* destination,
+                              const uint8_t* values, int64_t offset, int64_t num_values) {
+    RETURN_NOT_OK(
+        destination->Append(reinterpret_cast<const CType*>(values) + offset, num_values));
```

Comment:
Currently accounting for the type size with this cast. Will have to see how to extend to (fixed)-binary-types.

Reply:
You can look here for inspiration: https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/copy_data_internal.h

…and frankly, you can consider just using those utilities directly! I forgot about that